c10/core/impl/copy_on_write_peer.cpp

```diff
@@ -0,0 +1,43 @@
+#include <c10/core/impl/copy_on_write_peer.h>
+
+#include <c10/core/Storage.h>
+#include <c10/core/StorageImpl.h>
+#include <c10/core/TensorImpl.h>
+#include <c10/core/impl/copy_on_write_simulator.h>
+#include <c10/util/Exception.h>
+
+#include <cassert>
+
+namespace c10::impl {
+
+/* static */ auto CopyOnWritePeer::get_generation(Storage const& storage)
+    -> std::uint32_t {
+  assert(storage);
```

Comment:
don't use assert, we don't properly build with it. `TORCH_INTERNAL_ASSERT` for internal asserts that are cheap

Reply:
Is there a reason we don't have a build with asserts enabled? I feel like I've been in a conversation about this multiple times at this point.

Reply:
The point of `TORCH_INTERNAL_ASSERT` is that we can have it on all the time, which makes it substantially more useful for catching bugs in our binary distributions which aren't going to have asserts enabled.

Reply:
all `assert` gone